lib/_stream_writable.js

```diff
@@ -107,6 +112,13 @@ function WritableState(options, stream) {
 
   // True if the error was already emitted and should not be thrown again
   this.errorEmitted = false;
+
+  // when `state._end()` was called
+  this.flushed = false;
+
+  // NOTE: added here to not pollute the prototype of the WritableStream and to
+  // avoid conflicts with user-land methods
+  this._end = options._end || _end;
```

Comment:
This *might* be worthwhile space optimization to avoid creating a new function per stream instance. However, it's probably best to explore that as a separate, follow-on PR.